cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90

```diff
@@ -1131,8 +1140,15 @@ subroutine ModelAdvance(gcomp, rc)
 
        write(restart_date,"(i4.4,a,i2.2,a,i2.2,a,i5.5)") yy, '-', mm, '-',dd,'-',tod
        write(restart_filename,'(4a)') trim(restart_dir), trim(restart_file), '.', trim(restart_date)
-    else
-       force_restart_now = .false.
+    endif
+
+    ! Handle end of run restart
+    if (restart_eor) then
+       call ESMF_ClockGetAlarm(clock, alarmname='alarm_stop', alarm=alarm, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+       if (ESMF_AlarmIsRinging(alarm, rc=rc)) force_restart_now = .true.
+       call ESMF_AlarmRingerOff( alarm, rc=rc )
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
```

Comment:
You're not checking the return code for the AlarmIsRinging call. You'll want the same if-then logic as for the restart_alarm check.

Also, how is this generating the correct file name, if you're not getting the timestamp?